src/algorithms/tracking/SecondaryVertexFinder.cc

```diff
@@ -0,0 +1,414 @@
+  LinearizerSec linearizerSec(linearizerConfigSec, std::move(unique_log));
+
+  //Staring multivertex fitter
+  // Set up deterministic annealing with user-defined temperatures
+  Acts::AnnealingUtility::Config annealingConfig;
+  annealingConfig.setOfTemperatures = {9., 1.0};
+  Acts::AnnealingUtility annealingUtility(annealingConfig);
+  // Setup the vertex fitter
+  ImpactPointEstimator::Config ipEstConfig(m_BField, propagatorSec);
+  ImpactPointEstimator ipEst(ipEstConfig);
+  VertexFitterSec::Config vertexFitterConfigSec(ipEst);
+
+  vertexFitterConfigSec.annealingTool     = annealingUtility;
+  vertexFitterConfigSec.minWeight         = m_cfg.minWeight;
+  vertexFitterConfigSec.maxDistToLinPoint = m_cfg.maxDistToLinPoint;
+  vertexFitterConfigSec.doSmoothing       = m_cfg.doSmoothing;
+  vertexFitterConfigSec.useTime           = m_cfg.useTime;
+  vertexFitterConfigSec.extractParameters.connect<&Acts::InputTrack::extractParameters>();
+  vertexFitterConfigSec.trackLinearizer.connect<&LinearizerSec::linearizeTrack>(&linearizerSec);
+  VertexFitterSec vertexFitterSec(std::move(vertexFitterConfigSec));
+
+  // Set up vertex seeder and finder
+  std::shared_ptr<const Acts::IVertexFinder> seeder;
+  seedFinder::Config seederConfig(trkDensity);
+  seederConfig.extractParameters.connect<&Acts::InputTrack::extractParameters>();
+  seeder = std::make_shared<seedFinder>(seedFinder::Config{seederConfig});
+
+  VertexFinderSec::Config vertexfinderConfigSec(std::move(vertexFitterSec), std::move(seeder),
+                                                std::move(ipEst), m_BField);
+
+  // The vertex finder state
+  // Set the initial variance of the 4D vertex position. Since time is on a
+  // numerical scale, we have to provide a greater value in the corresponding
+  // dimension.
+  vertexfinderConfigSec.initialVariances = m_cfg.initialVariances;
+  //Use time for Sec. Vertex
+  vertexfinderConfigSec.useTime                    = m_cfg.useTime;
+  vertexfinderConfigSec.useSeedConstraint          = m_cfg.useSeedConstraint;
+  vertexfinderConfigSec.tracksMaxZinterval         = m_cfg.tracksMaxZintervalSec;
+  vertexfinderConfigSec.maxIterations              = m_cfg.maxSecIterations;
+  vertexfinderConfigSec.doFullSplitting            = m_cfg.doFullSplitting;
+  vertexfinderConfigSec.tracksMaxSignificance      = m_cfg.tracksMaxSignificance;
+  vertexfinderConfigSec.maxMergeVertexSignificance = m_cfg.maxMergeVertexSignificance;
+
+  if (m_cfg.useTime) {
+    // When using time, we have an extra contribution to the chi2 by the time
+    // coordinate.
+    vertexfinderConfigSec.tracksMaxSignificance      = m_cfg.tracksMaxSignificance;
+    vertexfinderConfigSec.maxMergeVertexSignificance = m_cfg.maxMergeVertexSignificance;
+  }
+
+  vertexfinderConfigSec.extractParameters.connect<&Acts::InputTrack::extractParameters>();
+  vertexfinderConfigSec.trackLinearizer.connect<&LinearizerSec::linearizeTrack>(&linearizerSec);
```

Comment:
Turns out this was not a thing

I don't see this in Acts example
https://github.com/acts-project/acts/blob/a8e9f5cb396c27a999b5bc052635581bf46063fc/Examples/Algorithms/Vertexing/src/AdaptiveMultiVertexFinderAlgorithm.cpp#L187-L188

```suggestion
```